### PR TITLE
Issue form GitHub actions

### LINKED ISSUE #2299: Create a new issue form for GitHub Actions
### Overview
As a developer, we want to standardize issue templates that are repeatedly used. For this issue, we will create an issue form template for GHA-related issues.

### Action Items

- [x] Create a gha-issue-template.yml file in [this folder](https://github.com/hackforla/website/tree/gh-pages/.github/ISSUE_TEMPLATE)
- [x] This template needs the following fields:
  - [x] Title
  - [x] Dependency (a check box to indicate yes/no dependency, then an input for what it is if yes)
  - [x] Overview
  - [x] Action Items/Psudo-code (optional)
  - [x] Checks (optional)
  - [x] Resources/Instructions
     - [x] a link to https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions
  - [x] Check box to indicate that this is to create a new GHA
    - [x] This checkbox, when checked, will add the staple instruction text (see below)
- [x] The resulting issue should look like this [issue](https://github.com/hackforla/website/issues/2206#issue-982258182)
  - [x] Be sure to pre-add the resources into this template

### Resources/Instructions
[Github Forms Issue Reading](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)
[Github Forms Issue Reading Pt 2](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema)

<details>
<summary>staple instruction text</summary>
<br>
Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)

Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!
<br>
</details>

### PR DESCRIPTION
**Pull request by arpitapandya**
Fixes #2299

### What changes did you make and why did you make them ?

  - created a gha-issue-template.yml file in [this folder](https://github.com/hackforla/website/tree/gh-pages/.github/ISSUE_TEMPLATE)
  - Form is visible

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
There aren't any changes to site.
- NOTE: I am not sure about the checkbox information I have used is correct.
- Time saver helpful tools: [extension to vs code](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml), [yamllint](http://www.yamllint.com/)

### CODE REVIEW

**Comment by Zak234** [comment]
ETA: Aug 7th
Availability: Aug 6th 9:00 AM - 3 PM

**Review by Zak234 — CHANGES_REQUESTED** [review]
There seems to be several issues with the template. First, there is no way for the user to add pseudo code when creating the issue as its just a label with no text field (also pseudo is spelled wrong). Second, I don't think there should be a description for textarea as the one included seems to be issue specific. Third, the label ``` New and existing unit tests pass locally with my changes``` should not be marked as required as this leads to the issue creator having to check the box to create the issue rather than the issue assignee. Fourth, the description ```Checklist on how to use the GitHub actions``` needs to be a label as it doesn't show up in the created issue currently.  Fifth, Action Items needs to be changed so the issue creator can actually add action items to the issue. Sixth, you cannot have ```"https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions"``` be a description as it doesn't appear in the created issue. Seventh, the ```Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)
        Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!``` description is supposed to be under resources.

**Comment by blulady** [comment]
> Fixes #2299
> 
> ### What changes did you make and why did you make them ?
> * created a gha-issue-template.yml file in [this folder](https://github.com/hackforla/website/tree/gh-pages/.github/ISSUE_TEMPLATE)
> * Form is visible
> 
> ### Screenshots of Proposed Changes Of The Website (if any, please do not screen shot code changes)
> There aren't any changes to site.
> 
> * NOTE: I am not sure about the checkbox information I have used is correct.
> * Time saver helpful tools: [extension to vs code](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml), [yamllint](http://www.yamllint.com/)

Thank you for sharing tools with us!! It really is delightful!!!

**Review by Zak234 — CHANGES_REQUESTED** [review]
There still seems to be an issue with the Resources-instructions not appearing upon creating the issue because it is a description. Additionally, the description for the Overview section seems to be specific to a single issue and doesn't seem like it applies to all Github Action issues. I also believe based on the posted example issue that pseudo code and action items are supposed to be separate sections.

**Comment by jdingeman** [comment]
ETA: EOD Aug 10
Availability: 4 hrs

File: .github/ISSUE_TEMPLATE/gha-issue-template.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        - [ ] I have made corresponding changes to the documentation
+        - [ ] My changes generate no lint errors
+        - [ ] I have added tests that prove my fix is effective or that my feature works
+        - [ ] New and existing unit tests pass locally with my changes
+  - type: textarea
+    id: Resources
+    attributes:
+      label: "Resources-instructions"
+      description: "Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)
+        Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!"
```
Comment (jdingeman) [inline .github/ISSUE_TEMPLATE/gha-issue-template.yml]: "Pseudo-code" is spelled wrong

**Review by jdingeman — CHANGES_REQUESTED** [review]
- Line 37: "Resources-instructions" should be "Resources/Instructions"
- Lines 18-24: Overview should be required.
- I agree with @Zak234 that Action Items and Pseudo-code should be separate sections
- The [original issue](https://github.com/hackforla/website/issues/2299) says that if the checkbox is checked, then it will display the staple text. I don't think there is a way in YAML to express conditional logic for our purposes. I think the staple text (the text containing the link to generic GitHub Actions) should appear with the last portion of the body. The Hack-for-LA's GitHub Actions should be in the description of Resources and Instructions.

**Comment by arpitapandya** [comment]
This PR is not ready to re-review as I am working on some changes, I have moved to Actively working for now. I will re-request once It's ready. Thank you

**Comment by arpitapandya** [comment]
Need help to delete the last commit.
How this happened: My intention was to make changes to the pull request. So, as I deleted the branch 2 days ago. I was trying to pull the branch locally so that I can work on it.
- command used `git checkout -b arpitapandya-issue-form-github-actions-2299 gh-pages`
- forked changes to my repo website for hackforla. I read few commits behind `update` so I forked changes
- which led me to this commit below I wanna get rid of
helpwanted

**Comment by jdingeman** [comment]
> Need help to delete the last commit. How this happened: My intention was to make changes to the pull request. So, as I deleted the branch 2 days ago. I was trying to pull the branch locally so that I can work on it.
> 
> * command used `git checkout -b arpitapandya-issue-form-github-actions-2299 gh-pages`
> * forked changes to my repo website for hackforla. I read few commits behind `update` so I forked changes
> * which led me to this commit below I wanna get rid of
>   helpwanted

Are you able to go into the previous commits and copy and paste the one you want it to revert back to? And then just commit it again?

**Comment by arpitapandya** [comment]
@Zak234 @jdingeman I have made changes, I am not sure if that even makes any sense. Do let me know, also if we still need to make changes to the last check box for staple text.

**Comment by arpitapandya** [comment]
Have to close this.

**Pull request closed without merging**